packages/components/src/templates/next/components/internal/SearchableTable/SearchableTableClient.tsx

```diff
@@ -0,0 +1,221 @@
+"use client"
+
+import { useDeferredValue, useId, useMemo, useRef, useState } from "react"
+
+import type { SearchableTableProps } from "~/interfaces"
+import { tv } from "~/lib/tv"
+import BaseParagraph from "../BaseParagraph"
+import { PaginationControls } from "../PaginationControls"
+import { SearchField } from "../Search"
+import { CellContent } from "./CellContent"
+import { MAX_NUMBER_OF_COLUMNS, PAGINATION_MAX_ITEMS } from "./constants"
+import { getFilteredItems } from "./getFilteredItems"
+import { getPaginatedItems } from "./getPaginatedItems"
+
+const createSearchableTableStyles = tv({
+  slots: {
+    container: "mx-auto w-full",
+    title: "prose-display-md break-words text-base-content-strong",
+    search: "mt-9",
+    tableContainer: "mt-8 overflow-x-auto",
+    table:
+      "[&_>_tbody_>_tr:nth-child(even)_>_td]:bg-base-canvas-default w-full border-collapse border-spacing-0 [&_>_tbody_>_tr:nth-child(odd)_>_td]:bg-base-canvas-alt",
+    tableRow: "text-left",
+    tableCell:
+      "max-w-40 break-words border border-base-divider-medium px-4 py-3 align-top last:max-w-full [&_li]:my-0 [&_li]:pl-1 [&_ol]:mt-0 [&_ol]:ps-5 [&_ul]:mt-0 [&_ul]:ps-5",
+    emptyState:
+      "flex flex-col items-center justify-center gap-8 self-stretch px-10 py-20 pt-24",
+    emptyStateHeadings: "text-center",
+    emptyStateTitle: "prose-headline-lg-regular",
+    emptyStateSubtitle:
+      "text-base-content-default prose-headline-lg-regular mt-3",
+    emptyStateButton:
+      "prose-headline-base-medium text-link hover:text-link-hover",
+    pagination: "mt-8 flex w-full justify-center lg:justify-end",
+  },
+  variants: {
+    isHeader: {
+      true: {
+        tableCell:
+          "bg-brand-interaction text-base-content-inverse [&_ol]:prose-label-md-medium [&_p]:prose-label-md-medium",
+      },
+      false: {
+        tableCell: "text-base-content [&_ol]:prose-body-sm [&_p]:prose-body-sm",
+      },
+    },
+    bold: {
+      true: {
+        emptyStateTitle: "text-base-content-strong",
+      },
+      false: {
+        emptyStateTitle: "text-base-content-subtle",
+      },
+    },
+  },
+})
+
+const compoundStyles = createSearchableTableStyles()
+
+export type SearchableTableClientProps = Omit<SearchableTableProps, "items"> & {
+  items: {
+    row: SearchableTableProps["items"][number]
+    key: string
+  }[]
+}
+
+export const SearchableTableClient = ({
+  title,
+  headers,
+  items,
+  sitemap,
+  LinkComponent,
+}: SearchableTableClientProps) => {
+  const [_search, setSearch] = useState("")
+  const search = useDeferredValue(_search)
+  const [currPage, setCurrPage] = useState(1)
+  const titleId = useId()
+
+  const maxNoOfColumns = Math.min(
+    headers.length,
+    ...items.map((row) => row.row.length),
```

Comment:
what happens if we have a case where the header length is shorter than the max row length? do we have to handle overflow? (never check)

Reply:
This checks every row and finds the minimum number of columns. So if we have header row with 6 columns, 1st row with 6 columns and 2nd row with 3 columns, the max we should display is 3.

Overflow is handled automatically via scroll.

Reply:
ah ok, so to double check what happens here - we only display 3 and there's a scrollbar allowing us to horizontally scroll and see the other 6? ok with this, i wanted to double check that we're not just removing the content